src/__tests__/CreatePlaylistButton.spec.jsx

```diff
@@ -0,0 +1,155 @@
+import React from "react";
+import {
+  render,
+  fireEvent,
+  waitFor,
+  screen,
+  act,
+} from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import CreatePlaylistButton from "../CreatePlaylistButton";
+import UserContext from "../contexts/UserContext";
+import useToken from "../hooks/useToken";
+import { addTracksToPlaylist, createPlaylist } from "../utils/spotify";
+import UnauthorizedError from "../utils/UnauthorizedError";
+
+jest.mock("../hooks/useToken");
+jest.mock("../utils/spotify");
+
+describe("tests for CreatePlaylistButton component", () => {
+  const customRender = (ui, { providerProps, ...renderOptions }) => {
+    return render(
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      <UserContext.Provider {...providerProps}>{ui}</UserContext.Provider>,
+      renderOptions
+    );
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useToken.mockImplementation(() => ["abc"]);
+  });
+
+  it("renders not disabled input for playlist name", () => {
+    const providerProps = {
```

Comment:
nit: to avoid repetitive object creation you can create a builder function, ie `buildProviderProps({onError: () => 'custom'})`